libvast/vast/detail/narrow.hpp

```diff
@@ -30,13 +30,11 @@
 
 namespace vast::detail {
 
-#if !VAST_NO_EXCEPTIONS
 /// @relates narrow
 struct narrowing_error : std::runtime_error {
   using super = std::runtime_error;
   using super::super;
 };
-#endif // VAST_NO_EXCEPTIONS
```

Comment:
Have you looked whether there are any `throw` statements guarded by this (or similar) defines?

Reply:
`git grep VAST_NO_EXCEPTIONS` comes up empty in this branch. There are a few `throw` statements in robin-map and function2, and one in the yaml wrapper in `data.cpp`,